### PR TITLE
[JENKINS-33819] New API to set the OrphanedItemStrategy field

### PR DESCRIPTION
**Pull request by recena**
[JENKINS-33819](https://issues.jenkins-ci.org/browse/JENKINS-33819)

@reviewbybees

### CODE REVIEW

**Comment by recena** [comment]
Deploying spanshots...

File: src/main/java/com/cloudbees/hudson/plugins/folder/computed/ComputedFolder.java [hunk 1]
```diff
@@ -108,6 +108,12 @@ protected ComputedFolder(ItemGroup parent, String name) {
         init();
     }
 
+    protected ComputedFolder(ItemGroup parent, String name, OrphanedItemStrategy strategy) {
```
Comment (jglick) [inline src/main/java/com/cloudbees/hudson/plugins/folder/computed/ComputedFolder.java]: :ant: Works for this one purpose, but more general and flexible to revert the new constructor and instead add `setOrphanedItemStrategy`.

Comment (amuniz) [inline src/main/java/com/cloudbees/hudson/plugins/folder/computed/ComputedFolder.java]: `orphanedItemStrategy` is commented as `/*almost final*/`, so I guess we don't want to modify it in a setter (not sure what was the reason to add that comment).

Comment (jglick) [inline src/main/java/com/cloudbees/hudson/plugins/folder/computed/ComputedFolder.java]: That comment is just wrong. `submit` sets it.

Comment (recena) [inline src/main/java/com/cloudbees/hudson/plugins/folder/computed/ComputedFolder.java]: @jglick To use `setOrphanedItemStrategy` was my first approach this weekend but while a discussed in private with @amuniz, I decided to change it, especially, for `/*almost final*/`.

**Comment by jglick** [comment]
:bee:

**Pull request merged**